### ISSUE TITLE
Amalgamation build not working for JS target.

### ISSUE THREAD

**Issue by Piyush3dB** [dmlc/mxnet#4909]
The following amalgamation command

```bash
make clean libmxnet_predict.js MIN=1
```

gives the error

```
In file included from mxnet_predict-all.cc:38:
emsdk_portable/emscripten/master/system/include/SSE/emmintrin.h:49:2: error: "SSE2 instruction set not enabled"
#error "SSE2 instruction set not enabled"
 ^
...
...
...
100 warnings and 2 errors generated.
ERROR:root:compiler frontend failed to generate LLVM bitcode, halting
make: *** [libmxnet_predict.js] Error 1
```

My Emscripten version is:

```
emcc -v
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 1.37.0
clang version 3.9.0 (https://github.com/kripken/emscripten-fastcomp-clang/ 60a7e9a9c22b67309e5b1258d38fadfa481a25d3) (https://github.com/kripken/emscripten-fastcomp/ 881bd352731d21c7117ad7e2ece347aacae83965) (emscripten 1.37.0 : 1.37.0)
Target: x86_64-unknown-linux-gnu
Thread model: posix
```

It seems emcc requires the macro `__SSE__` to be defined otherwise there is an error is reported by the line in `emmintrin.h`

The last known working version of amalgamation is in the mxnet.js repo is over a year old.  Has anyone successfully built a more recent MXNet version for the JS target?

**Comment by Piyush3dB** [comment]
@SlipknotTN have you seen this before?

**Comment by SlipknotTN** [comment]
I didn't follow mxnet latest months and I see very recent commits in amalgation part. However you can try to remove #include <emmintrin.h> if it is present in your mxnet_predict-all.cc .
I managed to build more recent version than mxnet js repo, more or less at the time when I modified the amalgation script through a pull request. But I didn't follow the subsequent changes.

**Comment by Piyush3dB** [comment]
Issue resolved in PR #5090. Closing.
